server/py/framework/db/sqldb/db.py

```diff
@@ -525,10 +526,20 @@ def del_runs(
     def _enrich_run_struct_from_model(
         self, run: Run, run_struct: dict, with_notifications: bool
     ):
-        if run.end_time:
-            run_struct.setdefault("status", {})["end_time"] = self._add_utc_timezone(
-                run.end_time
-            ).isoformat()
+        status = run_struct.setdefault("status", {})
+
+        # These fields are saved in struct as timestamps with fsp=6, while the corresponding columns
+        # in the database have fsp=3. Since 'ORDER BY' is applied to the column, we return the value from
+        # the column (not from the struct) to ensure the ordering is correct.
+        # In SQLite, the start_time and updated columns return timestamps with fsp=6.
```

Comment:
Comment is not accurate because end time is a datetime and is fsp 6
```suggestion
        # Return the value from the column to ensure the ordering is correct since the sort is done on the table columns and
        # timestamps are being saved with fsp=3 while struct fields are fsp=6.
        # In SQLite, the start_time and updated columns return timestamps with fsp=6.
```